### PR TITLE
chore(menu-button): documentation, add testid to wrapper

### LINKED ISSUE #2265: [MenuButton] Testing
MenuButton component needs comprehensive testing in a few areas. This issue is considered complete when:

- [x] data test ids have been provided to appropriate pieces 
- [x] Unit test are 80%+
- [ ] a11y scans in storybook do not show any violations 
- [x] a11y unit test has been written using the IBM scan tool
- [x] RTL has been tested and working properly. 
- [x] Confirm that all strings are able to be substituted for i18n

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #2265 

Depends on https://github.com/carbon-design-system/carbon/pull/8561 for a11y tests to pass.

**Summary**

- added data-testids to various components,
- add documentation using withReadme wrapper
- add extra test to cover getMenuPositioning for autopositioning
- add a11y test.
- added knob to change which component gets used on the auto-positioning story

**Acceptance Test (how to verify the PR)**

- All tests pass

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Menu functions in storybook

### CODE REVIEW

File: packages/react/src/components/MenuButton/utils.js [hunk 1]
```diff
@@ -36,50 +36,62 @@ export const getMenuPosition = ({ label, buttonRef, onPrimaryActionClick, langDi
...
       case 'top':
         y = buttonY + buttonHeight;
-        x = buttonX;
+        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
         break;
       case 'left':
-        x = buttonX;
+        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
         y = buttonY + buttonHeight;
         break;
```
Comment (herleraja) [inline packages/react/src/components/MenuButton/utils.js:81]: merge the similar cases. 

```
 case 'top':
 case 'left':
 default:
        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
        y = buttonY + buttonHeight;
        break;
```

File: packages/react/src/components/MenuButton/utils.js [hunk 2]
```diff
@@ -36,50 +36,62 @@ export const getMenuPosition = ({ label, buttonRef, onPrimaryActionClick, langDi
...
+          ? isRtl
+            ? buttonX + buttonWidth + primaryButtonWidth
+            : buttonX + buttonWidth
+          : buttonX + buttonWidth;
         y = buttonY - menuHeight;
         break;
       case 'bottom-left':
-        x = buttonX;
+        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
         y = buttonY - menuHeight;
```
Comment (herleraja) [inline packages/react/src/components/MenuButton/utils.js:72]: case `bottom` and `bottom-left` seem to be similar so merge the cases.

**Review by herleraja — CHANGES_REQUESTED** [review]
LGTM. Need some cleaning in switch case.

File: packages/react/src/components/MenuButton/utils.js (hunk 2 above)
Comment (kevinsperrine) [reply packages/react/src/components/MenuButton/utils.js]: thanks, @herleraja. so many possible conditions, I was just trying to get everything covered. Cleaned it up now.

**Review by herleraja — APPROVED** [review]
LGTM

**Review by davidicus — APPROVED** [review]

**Pull request merged**